src/resource.rs

```diff
@@ -113,5 +130,11 @@ pub trait WriteAtomic: io::Write {
     /// with a system-level error.
     ///
     /// This method shouldn't be called directly; [`Self::write_atomic`] must be used instead.
+    ///
+    /// # Safety
+    ///
+    /// The method must handle EGAGAIN, EINTER, EWOULDBLOCK I/O errors and buffer the data in such
+    /// cases. Ig these errors are returned from this methods [`WriteAtomic::write_atomic`] will
+    /// panic.
```

Comment:
Interesting. You don't want to handle those as regular errors? I guess it only panics in `debug` mode according to the above `debug_assert`? Should that be specified here?

Reply:
The reply to the comment above applies here as well